pywbem/_logging.py

```diff
@@ -32,53 +32,103 @@
+users of pywbem that want simply one or two line setup for logs as follows:
+
+* :meth:`~pywbem._logging.define_loggers_from_string` defines one or more
+  loggers from  a string input that defines the component name and
+  characteristics of each logger.  This allows other tools like CLIs that use
+  pywbem to create the pywbem known logs with minimal work from command line or
+  config file input.
+
+* :meth:`~pywbem._logging.define_logger` defines  one
+  logger from parameter inputs that define the component name and
+  characteristics of the logger.
+
+Thus to setup loggers that log all of the operations requests and responses
+both for http and the APIs either of the following could be use::
+
+    # Log ops to stderr and http to file but only log the first n characters
+    # where n is defined by the pywbem variable `DEFAULT_MAX_LOG_ENTRY_SIZE`
+    define_logger('ops', 'stderr')
+    define_logger('http', 'file', log_filename='mylog.log')
+    conn = WBEMConnection(..., enable_log=min)
+
+    # Log everything to stderr and output all of the information
+    define_loggers_from_string('all=stderr')
+    conn = WBEMConnection(..., enable_log='all')
+
+    # Log only http requests and responses to a file
+    define_logger('http', 'file', log_filename='mylog.log')
+    conn = WBEMConnection(..., enable_log=min)
+
+The pywbem loggers are based one parameters (log destination)
 that determine if the logs for the logger name are created,
 how much information is inserted, and the log destination.  This
 extends the python logging facility to include the log_detail parameter which
 defines whether all the information defined or only a limited size is output.
 This is used with pywbem because the logs on operation and http responses can be
 very large.
 
-The code that executes the loggers call the function get_logger(..) to get a
-logger from a defined logger name.  If that logger has not yet been defined
-in PywbemLoggers, an entry will be added with the default parameters.
+Logging may be define differently for different WBEMConnections by adding
+the WBEMConnection id as a part of the logger component.  Thus::
+
+    define_logger('ops', 'stderr')
+    define_logger('http', 'file', log_filename='mylog.log')
+    conn1 = WBEMConnection(..., enable_log="min")
+
+    conn2 = WBEMConnection(..., enable_log='all")
+
+    logger = logging.getLogger('pywbem.ops.%s' % conn2.conn_id)
+    handler = logging.FileHandler('logfilename.log')
+    handler.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(message)s')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+
+Since logging can generate extensive output, the :class:`~pywbem:WBEMConnection`
+allows the user to both activate the defined logging and set log output
+size characteristics.
+
+If logging is activated without any loggers being defined the
+logging NULLHandler is set for both the operations and http loggers.
```

Comment:
In the documentation, this module doc string is followed immediately by the various constants defined in the module. I wonder which of these are actually needed to become part of the external API. Also, some introductory text about the nature of them would be helpful.